sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -97,18 +97,54 @@ object GpuParquetFileFormat {
     }
   }
 
-  def getPrecisionList(schema: StructType): Seq[Int] =  {
-    def precisionsList(t: DataType): Seq[Int] = {
-      t match {
-        case d: DecimalType => List(d.precision)
-        case _: StructType =>
-          throw new IllegalStateException("structs are not supported right now")
-        case _: ArrayType =>
-          throw new IllegalStateException("arrays are not supported right now")
-        case _ => List(0)
+  def parquetWriterOptionsHelper(schema: StructType): ParquetWriterOptions.Builder[_] = {
+    def addColumnMeta(name: String, nullable: Boolean,
```

Comment:
I'm not a fan of internal functions that are quite sizeable, as they get in the way of reading the function they are in.  I suggest moving this out as a private peer method.